src/main/kotlin/org/rust/cargo/project/settings/impl/SerializationUtils.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+@file:JvmName("SerializationUtils")
+@file:Suppress("unused")
+
+package org.rust.cargo.project.settings.impl
+
+import com.intellij.util.addOptionTag
+import com.intellij.util.xmlb.Constants
+import org.jdom.Element
+import org.rust.cargo.project.settings.RustProjectSettingsService.MacroExpansionEngine
+
+// Bump this number if Rust project settings changes
+const val XML_FORMAT_VERSION: Int = 3
+
+const val VERSION: String = "version"
+const val TOOLCHAIN_HOME_DIRECTORY: String = "toolchainHomeDirectory"
+const val AUTO_UPDATE_ENABLED: String = "autoUpdateEnabled"
+const val EXPLICIT_PATH_TO_STDLIB: String = "explicitPathToStdlib"
+const val EXTERNAL_LINTER: String = "externalLinter"
+const val RUN_EXTERNAL_LINTER_ON_THE_FLY: String = "runExternalLinterOnTheFly"
+const val EXTERNAL_LINTER_ARGUMENTS: String = "externalLinterArguments"
+const val COMPILE_ALL_TARGETS: String = "compileAllTargets"
+const val USE_OFFLINE: String = "useOffline"
+const val MACRO_EXPANSION_ENGINE: String = "macroExpansionEngine"
+const val SHOW_TEST_TOOL_WINDOW: String = "showTestToolWindow"
+const val DOCTEST_INJECTION_ENABLED: String = "doctestInjectionEnabled"
+const val RUN_RUSTFMT_ON_SAVE: String = "runRustfmtOnSave"
+const val USE_SKIP_CHILDREN: String = "useSkipChildren"
+
+// Legacy properties needed for migration
+const val USE_CARGO_CHECK_ANNOTATOR: String = "useCargoCheckAnnotator"
+const val CARGO_CHECK_ARGUMENTS: String = "cargoCheckArguments"
+const val EXPAND_MACROS: String = "expandMacros"
+
+fun Element.updateToCurrentVersion() {
+    updateToVersionIfNeeded(2) {
+        renameOption(USE_CARGO_CHECK_ANNOTATOR, RUN_EXTERNAL_LINTER_ON_THE_FLY)
```

Comment:
Hm, we already can have users with current weird migration scheme so we can't just rename option
A user can already have `RUN_EXTERNAL_LINTER_ON_THE_FLY` field and this renaming creates a new option with the same name

Reply:
But a user cannot have `USE_CARGO_CHECK_ANNOTATOR` and `RUN_EXTERNAL_LINTER_ON_THE_FLY` at the same time because we set the `useCargoCheckAnnotator` to default:
https://github.com/intellij-rust/intellij-rust/blob/5b9068d6f7d8aa4fb4edc13af4fed1862116dfcd/src/main/kotlin/org/rust/cargo/project/settings/impl/RustProjectSettingsServiceImpl.kt#L43-L46

Reply:
Oh, yeah. You are right